galley/pkg/runtime/monitoring/monitoring.go

```diff
@@ -201,18 +210,19 @@ func getStateTypeConfigKeys() ([]tag.Key, error) {
 	return []tag.Key{NamespaceTag, NameTag, VersionTag}, err
 }
 
-func registerNewStateTypeConfigView(collection string) error {
-	stateTypeConfigTotal[collection] = stats.Int64(fmt.Sprintf("galley/runtime/state/%s_total", collection),
+func registerNewStateTypeConfigView(collection string) (*stats.Int64Measure, error) {
+	measure := stats.Int64(fmt.Sprintf("galley/runtime/state/%s_total", collection),
 		fmt.Sprintf("The number of valid %v known to galley at a point in time", collection),
 		stats.UnitDimensionless)
+	stateTypeConfigTotal.Store(collection, measure)
 	nameKeys, err := getStateTypeConfigKeys()
 	if err != nil {
-		return err
+		return measure, err
 	}
 	err = view.Register(
-		newView(stateTypeConfigTotal[collection], nameKeys, view.LastValue()),
+		newView(measure, nameKeys, view.LastValue()),
```

Comment:
This is still racy. There is nothing stopping to racing go routines from calling view.Register with the same view. This will cause an error to be returned.

Reply:
Thanks for checking.. somehow read the error log from tryregisterview incorrectly...